semgrep-core/parsing/Parse_code.ml

```diff
@@ -76,7 +78,7 @@ let just_parse_with_lang lang file =
               Ast_js_build.program cst
             )
             file
-        with _exn -> Parse_javascript_tree_sitter.parse file
+        with ForceTreeSitter -> Parse_javascript_tree_sitter.parse file
```

Comment:
We want to try treesitter if there is any exn in the pfff parser (lexer exn, parse error) or when we have -forcetreesitter

Reply:
So it should remain _exn here

Reply:
You and @mjambon disagree here...

Reply:
The idea is to not print what went wrong when the error is a parse error in the pfff parser.
We should not bother the user with that. We first try with pfff, and if it fails we try with tree-sitter (or the reverse, it depends on the language).
But I'll change the option to -tree_sitter_only, maybe it's clearer.